core/src/main/java/org/apache/spark/shuffle/unsafe/UnsafeShuffleWriter.java

```diff
@@ -147,7 +150,7 @@ public void write(scala.collection.Iterator<Product2<K, V>> records) throws IOEx
       closeAndWriteOutput();
       success = true;
     } finally {
-      if (!success) {
+      if (!success && sorter != null) {
```

Comment:
this isn't necessary for this patch, but it prevents an NPE if something goes wrong in `closeAndWriteOutput` after `sorter` is set to `null`.  (eg., when I was working on this and introduced a bug in `closeAndWriteOutput` that I needed to debug, the actual error was swallowed and instead i just got an NPE here)